libsql-server/src/namespace/mod.rs

```diff
@@ -190,110 +78,73 @@ pub enum NamespaceBottomlessDbIdInit {
     FetchFromConfig,
 }
 
-/// Creates a new `Namespace` for database of the `Self::Database` type.
-#[async_trait::async_trait]
-pub trait MakeNamespace: Sync + Send + 'static {
-    type Database: Database;
+/// A namespace isolates the resources pertaining to a database of type T
+#[derive(Debug)]
+pub struct Namespace {
+    pub db: Database,
+    name: NamespaceName,
+    /// The set of tasks associated with this namespace
+    tasks: JoinSet<anyhow::Result<()>>,
+    stats: Arc<Stats>,
+    db_config_store: MetaStoreHandle,
+}
 
-    /// Create a new Namespace instance
-    async fn create(
-        &self,
-        name: NamespaceName,
+impl Namespace {
+    async fn from_config(
+        ns_config: &NamespaceConfig,
+        db_config: MetaStoreHandle,
         restore_option: RestoreOption,
-        bottomless_db_id: NamespaceBottomlessDbId,
```

Comment:
What happened to this param?

Reply:
let's track this params down.
- it's passes to try [`try_new_primary`](https://github.com/tursodatabase/libsql/blob/main/libsql-server/src/namespace/mod.rs#L1096-L1109), where it's either passed as `Namespace` or read from config. `try_new_primary` is called by [`new_primary`](https://github.com/tursodatabase/libsql/blob/main/libsql-server/src/namespace/mod.rs#L1054), which in turn is called by [`PrimaryNamespaceMaker::create`](https://github.com/tursodatabase/libsql/blob/main/libsql-server/src/namespace/mod.rs#L252-L259)

- `MakeNamespace::create` is called by:
    - [`with`](https://github.com/tursodatabase/libsql/blob/main/libsql-server/src/namespace/mod.rs#L669-L676) (no provided)
    - [`reset`](https://github.com/tursodatabase/libsql/blob/main/libsql-server/src/namespace/mod.rs#L535-L541) (not provided)
    - in [`fork`](https://github.com/tursodatabase/libsql/blob/main/libsql-server/src/namespace/mod.rs#L605-L612) (not provided)
    - in [`try_fork`](https://github.com/tursodatabase/libsql/blob/main/libsql-server/src/namespace/fork.rs#L109-L120), from `self.bottomless_id`, taken from [source namespace config](https://github.com/tursodatabase/libsql/blob/main/libsql-server/src/namespace/mod.rs#L314)
    - from `[NamespaceStore::create`](https://github.com/tursodatabase/libsql/blob/main/libsql-server/src/namespace/mod.rs#L742-L749), only called by [`HTTP::admin::handle_create`](https://github.com/tursodatabase/libsql/blob/main/libsql-server/src/http/admin/mod.rs#L332), with the param from the request. 
    
  So it's only dependent on the value of the if provided during namespace creation. That value was not stored in the meta, and passed directly to namespace creation. If you ask me this whole code was smelly AF. Instead, we set the db id in the config, and create the namespace with a consistent config.

Reply:
So we just read it from config.